### ISSUE TITLE
Ubuntu: Deb, Snap and Flatpak dont work with the GUI suddenly

### ISSUE THREAD

**Issue by firefoxlover** [onionshare/onionshare#1601]
I only tried the Deb package, the others didnt open the GUI through the Appstartet too

```
user@laptop:~$ onionshare-gui
OnionShare 2.2 | https://onionshare.org/
Traceback (most recent call last):
  File "/usr/bin/onionshare-gui", line 22, in <module>
    onionshare_gui.main()
  File "/usr/lib/python3/dist-packages/onionshare_gui/__init__.py", line 142, in main
    gui = OnionShareGui(common, onion, qtapp, app, filenames, config, local_only)
  File "/usr/lib/python3/dist-packages/onionshare_gui/onionshare_gui.py", line 172, in __init__
    self.share_mode.init()
  File "/usr/lib/python3/dist-packages/onionshare_gui/mode/share_mode/__init__.py", line 48, in init
    self.web = Web(self.common, True, "share")
  File "/usr/lib/python3/dist-packages/onionshare/web/web.py", line 77, in __init__
    self.generate_static_url_path()
  File "/usr/lib/python3/dist-packages/onionshare/web/web.py", line 166, in generate_static_url_path
    self.app.add_url_rule(
  File "/usr/lib/python3/dist-packages/flask/scaffold.py", line 56, in wrapper_func
    return f(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/flask/app.py", line 1092, in add_url_rule
    raise AssertionError(
AssertionError: View function mapping is overwriting an existing endpoint function: static
```

**Comment by mig5** [comment]
I think this issue was fixed long ago https://github.com/onionshare/onionshare/issues/1365

But we don't really make .debs any more (the PPA is no longer active), so maybe you're on an old version.. (the .deb in Debian is very old)

In any case, I would suggest to await the next release and then try either the Snap or Flatpak instead of .deb.

**Comment by Hxs2fJKjYZ** [comment]
The flatpak version doesn't work on Fedora 36 either. It tries to load but then fails, no window opens.

**Comment by xznhj8129** [comment]
Same problem, fails to open with "ImportError: cannot import name 'soft_unicode' from 'markupsafe' "

**Comment by joerg-knitter** [comment]
Same error here on DietPi (Debian-based).

**Comment by hopeless-attempt** [comment]
Same problem, fails to open with "ImportError: cannot import name 'soft_unicode' from 'markupsafe' "
Mx-Linux 21.1 KDE
Latest Flatpak Onionshare 2.5

**Comment by Shmeckser** [comment]
```
  File "/app/lib/python3.9/site-packages/jinja2/filters.py", line 13, in <module>
    from markupsafe import soft_unicode
ImportError: cannot import name 'soft_unicode' from 'markupsafe' (/usr/lib/python3.9/site-packages/markupsafe/__init__.py)
```
I have a similar issue on newest flatpak on Linux Mint.

**Comment by jaykayenn** [comment]
'soft_unicode' has been deprecated from MarkupSafe: https://markupsafe.palletsprojects.com/en/2.1.x/changes/

**Comment by Shmeckser** [comment]
> 'soft_unicode' has been deprecated from MarkupSafe: https://markupsafe.palletsprojects.com/en/2.1.x/changes/

Yeah, but even when I downgraded Markupsafe to i think 2.0.1 where 'soft_unicode' wasn't deprecated yet, it still didnt work. And maybe this is a flatpak thing but the Importerror points to a direction that doesn't exist since i have python3.10.

**Comment by mig5** [comment]
If you searched the issue queue you would find all these issues talking about the same thing:

https://github.com/onionshare/onionshare/issues/1536
https://github.com/onionshare/onionshare/issues/1608
https://github.com/onionshare/onionshare/issues/1606
https://github.com/onionshare/onionshare/issues/1605
https://github.com/onionshare/onionshare/issues/1600
https://github.com/onionshare/onionshare/issues/1597
https://github.com/onionshare/onionshare/issues/1585
https://github.com/onionshare/onionshare/issues/1571

TLDR: It has been fixed in git (https://github.com/onionshare/onionshare/pull/1541) but there has not yet been a new official release that contains the fix. The next Flatpak release will contain the fix. There is no estimate of when that will be.

**Comment by Shmeckser** [comment]
Ok, thank you for showing how to remedy the issue.

**Comment by kazin-essen** [comment]
> If you searched the issue queue you would find all these issues talking about the same thing (also you might note that `'Same problem, fails to open with "ImportError: cannot import name 'soft_unicode' from 'markupsafe'` is not actually the same problem as the error mentioned in the start of this issue, so not sure why we're talking about it here)
> 
> #1536 #1608 #1606 #1605 #1600 #1597 #1585 #1571
> 
> TLDR: The markupsafe/soft_unicode issue has been fixed in git (#1541) since March, but there has not yet been a new official release that contains the fix. The next Flatpak release will contain the fix. There is no estimate of when that will be.

Well, it would be nice if this could be appended to the main install page for users to know what is going on and how to fix this. We thank the contributors of course but unconscious bias does often prevent one from seeing that not all may realise to dig this deep.

**Comment by mig5** [comment]
You are right, we could and should have put something on the website since there seem to be so many Flatpak users. We pinned an issue at the top of the issue queue, but it seems to be missed.

The good news is we are about to make a new release and just today Micah got Flatpak building it successfully. It won't be long now. Thanks for your patience, it's been an absurdly long wait, just due to developer availability.

**Comment by kazin-essen** [comment]
> You are right, we could and should have put something on the website since there seem to be so many Flatpak users. We pinned an issue at the top of the issue queue, but it seems to be missed.
> 
> The good news is we are about to make a new release and just today Micah got Flatpak building it successfully. It won't be long now. Thanks for your patience, it's been an absurdly long wait, just due to developer availability.

I must say, your response impressed me my friend. You tok responsibility and didn’t point fingers. Unconscious bias is something I am all too familiar with as a technical writer and so I know that none of it comes from ill intent; still, I have to often dissect the tech talk to the Everyman and well, you know the tedium. :P

I am please to hear about the new release. Open source projects are labour of love and so we can’t demand swiftness or perfection, only empathy. :)

**Comment by mig5** [comment]
Hopefully all fixed with https://github.com/onionshare/onionshare/releases/tag/v2.6

**Comment by Asta1986** [comment]
With v2.6 the Flatpak is working again. Thanks for fixing this.

**Comment by Hxs2fJKjYZ** [comment]
Still isn't working for me on Fedora 36. It seemed to load up okay for the first time, then after that I got a Python 3.9 crash and now the app won't load again and there are no error/crash messages after the initial Python 3.9 crash message. The snap package is working which will have to do for now, but I do prefer to have flatpak than snap.

**Comment by mig5** [comment]
It's not enough to know if this crash relates to the issue described here. We would need to know more about the crash message.

Regarding the 'app won't load again' - yes, annoying issue. Look in `~/.config/onionshare` for a lock file and if you find it, delete it. Hopefully you can then start it again (and maybe see what that crash message is). See https://github.com/onionshare/onionshare/issues/1374

**Comment by Hxs2fJKjYZ** [comment]
There was no lock file in the folder, so I uninstalled the app, then reinstalled it, still doesn't load up and no error or crash messages, pretty much the same issue as before the 2.6 update. I tried running from command line, screenshot attached.
Looks like it's linked to the thread you sent?
Is it possible the lock file could be stored in another location for the flatpak version of the app?
![Screenshot from 2022-10-14 10-12-21](https://user-images.githubusercontent.com/12845068/195746614-84f603ea-f5ef-49ee-ae19-bf1520aba530.png)

**Comment by mig5** [comment]
Yep, that's definitely the 'lock file exists' scenario. On Flatpak, looks like it might be at `~/.var/app/org.onionshare.OnionShare/config/onionshare/lock`

**Comment by Hxs2fJKjYZ** [comment]
Found it under ~/.var/app/org.onionshare.OnionShare/config/onionshare
I deleted the file and onionshare opened, with about a dozen tabs, I closed all the tabs, then closed onionshare and got the Python 3.9 crash again and the lock file persisted. so I deleted the lock file again and tried to launch, but received the Python crash again and the app crashed with it this time.
How to post the Python crash data here?
